R/guide-legend.R

```diff
@@ -559,9 +560,16 @@ GuideLegend <- ggproto(
       gt <- gtable_add_grob(
         gt, elements$background,
         name = "background", clip = "off",
-        t = 1, r = -1, b = -1, l =1, z = -Inf
+        t = 1, r = -1, b = -1, l = 1, z = -Inf
       )
     }
+
+    # for inside guide legends, we also save the position values 
+    # in this way, we can identify legends with same position
+    # and then merge them into same guide box in `Guides$draw()`
+    if (identical(.subset2(params, "position"), "inside")) {
+      attr(gt, "inside_position") <- .subset2(elements, "inside_position")
+    }
```

Comment:
If think it should be possible to circumvent touching the legend/colourbar guides here if we move away from this attribute-based propagation of inside coordinates. If we can make this work without changing the guides, it will work for `guide_custom()` and other extension guides too.

Reply:
All modifications to guide_legend and guide_colorbar have now been reverted.